hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelinePlacementPolicy.java

```diff
@@ -124,6 +124,15 @@ List<DatanodeDetails> filterViableNodes(
     // get nodes in HEALTHY state
     List<DatanodeDetails> healthyNodes =
         nodeManager.getNodes(NodeStatus.inServiceHealthy());
+    if (healthyNodes.size() < nodesRequired) {
```

Comment:
We are checking this at multiple places here itself. For eg, We reassign `healthyNodes` with `filterNodesWithSpace()`, itself has check for `nodesWithSpace` which is returned back so theoretically `nodesWithSpace <= healthyNodes`.
https://github.com/apache/ozone/blob/1ff97328b7de88bc35c1665f1c90c0782b3cf7e2/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/SCMCommonPlacementPolicy.java#L226-L242

I don't think this is fixing the problem, I tried running the test with just modifying this line, and it worked.
https://github.com/apache/ozone/blob/4454e6b368bc5aedaba509eeb85422daeda393bf/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestRatisPipelineCreateAndDestroy.java#L148-L149

Reply:
Existing code is not checking required condition healthyNodes.size() < nodesRequired (where new check was added in the current change) before calling filterNodesWithSpace.

Reply:
With the current fix , it is checking condition before calling filterNodesWithSpace, and throwing exception with code FAILED_TO_FIND_SUITABLE_NODE.

Test is passing.

Reply:
Two others alternative fixes are working fine  

 1) changing expected code to FAILED_TO_FIND_NODES_WITH_SPACE in test

https://github.com/apache/ozone/blob/4454e6b368bc5aedaba509eeb85422daeda393bf/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestRatisPipelineCreateAndDestroy.java#L148


     But is it ok to expect FAILED_TO_FIND_NODES_WITH_SPACE after shutting down all data nodes ?

2) Removing assertEquals on result code

Reply:
@kerneltime, this patch looks to correct the failing disabled integration test, `TestRatisPipelineCreateAndDestroy.testPipelineCreationOnNodeRestart.`  The test fails because the exception thrown in the test differs from the expected exception.  The patch ensures the expected exception is thrown.  @pratapchandu, thanks for filing the PR.  Updating the Jira with a brief description of the error would also be helpful.

Reply:
I don't think this assertion was the issue previously, 
https://github.com/apache/ozone/blob/4454e6b368bc5aedaba509eeb85422daeda393bf/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestRatisPipelineCreateAndDestroy.java#L148
This issue was created long back, I think it was fixed by subsequent code changes over the year. 

Even without adding this change, it would have thrown an exception as `nodesWithSpace` is a subset of `healthyNodes`. So I don't think we need to be checking it again. As for the test change I did this,

> 1. changing expected code to FAILED_TO_FIND_NODES_WITH_SPACE in test

Reply:
@aswinshakil  thanks for review , resolving this